packages/react-tc-ui-kit/src/components/Textarea/Textarea.js

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useRef } from 'react';
+
+import './Textarea.css';
+
+const DEFAULT_TEXTAREA_HEIGHT = 120;
+
+export default function Textarea({
+  inputRef,
+  style,
+  id,
+  className = '',
+  label,
+  disabled,
+  value = '',
+  maxLength,
+  autoSize = false,
+  placeholder,
+  isValid,
+  isInvalid = false,
+  validationMessages = [],
+  isMessagesAbsolute = false,
+  onChange = () => {},
+  ...props
+}) {
+  const textareaRef = inputRef || useRef();
+
+  useEffect(() => {
+    handleResize(textareaRef.current);
+  }, []);
+
+  const validClassname = isValid ? 'tc-textarea--valid' : '';
+  const invalidClassname = isInvalid ? 'tc-textarea--invalid' : '';
+  const errorsAbsoluteClassname = isMessagesAbsolute ? 'tc-textarea__errors--absolute' : '';
+  const autosizeClassname = autoSize ? 'tc-textarea__control--autosize' : '';
+
+  return (
+    <div
+      style={style}
+      className={`tc-textarea ${className} ${invalidClassname} ${validClassname}`}
+    >
+      {label && (
+        <label className="tc-textarea__label" htmlFor={id}>{label}</label>
+      )}
+
+      <div className="tc-textarea__box">
+        <textarea
+          ref={textareaRef}
+          id={id}
+          maxLength={maxLength}
+          placeholder={placeholder}
+          disabled={disabled}
+          className={`tc-textarea__control ${className} ${autosizeClassname}`}
+          value={value}
+          onChange={handleChange}
+          {...props}
+        />
+
+        <div className="tc-textarea__control-counter">
+          {value.length}
```

Comment:
we should add some `null` check here
in case if value will be null or undefined